doc/source/guide/derivatives_guide.rst

```diff
@@ -52,25 +57,31 @@ Here :math:`f'(x)` is defined as the linear operator (if it exists) that satisfi
 
 This definition of the derivative is called the *Fréchet derivative*.
 
+The Gateaux derivative
+~~~~~~~~~~~~~~~~~~~~~~
+The concept of directional derivative can also be extended to Banach spaces, giving the *Gateaux* derivative. 
+The Gateaux derivative is more general than the Fréchet derivative, but is not a linear operator. An example of a function that is Gateaux differentiable but not Fréchet differentiable is the absolute value function.
+It not used in ODL since it can fail to be linear. 
```

Comment:
Hm, actually we allow for both concepts. Users can define an operator with non-linear derivative. However, substantial amounts of code (mostly solvers I guess) assume that the derivative is linear since it's used in the form `op.derivative(point).adjoint`. But as long as such a construction is not used, non-linear Gateaux derivatives are also supported.